submissions/team_api.py

```diff
@@ -137,7 +137,20 @@ def create_submission_for_team(
         'item_id': item_id,
         'item_type': item_type
     }
+
+    students_with_team_submissions = [submission.student_item.student_id for submission in Submission.objects.filter(
```

Comment:
I think this can be replaced? You're querying on Submission, but you're really lookup up StudentItems. 

```
[student_item.student_id for student_item in StudentItem.objects.filter(
        submission_set__team_submission__isnull=False
    ).exclude(
        submission_set___team_submission__team_id=team_id
    ).filter(
        student_id__in=team_member_ids,
        course_id=course_id,
        item_id=item_id
    )]

```

Reply:
Also, you have `select_related('team_submission')` but it doesn't look like you're navigating through that